hyp3_gamma/insar/unwrapping_geocoding.py

```diff
@@ -218,20 +220,22 @@ def unwrapping_geocoding(reference, secondary, step="man", rlooks=10, alooks=2,
 
     execute(f"rasmph_pwr {ifgf}.adf {mmli} {width}", uselogging=True)
 
-    execute(f"rascc_mask {ifgname}.adf.cc {mmli} {width} 1 1 0 1 1 0.10 0.20 ", uselogging=True)
-
     if apply_water_mask:
-        # create and apply water mask
-        out_file = combine_water_mask(f"{ifgname}.adf.cc_mask.bmp", mwidth, mlines, lt, demw, demn, dempar)
+        # produce water mask and apply to {ifgname}.adf.cc and {mmli} files
+        adf_cc_masked, _ = get_masked_files(f"{ifgname}.adf.cc", mmli, int(mwidth), int(mlines), lt, int(demw),
+                                            int(demn), dempar)
     else:
-        # create water mask only
-        _ = get_water_mask(f"{ifgname}.adf.cc_mask.bmp", mwidth, lt, demw, demn, dempar)
-        out_file = f"{ifgname}.adf.cc_mask.bmp"
+        # produce water mask file only
+        _ = get_water_mask(f"{ifgname}.adf.cc", mwidth, lt, demw, demn, dempar)
+        adf_cc_masked = f"{ifgname}.adf.cc"
```

Comment:
Overall, the logic would be a little clearer if this section worked like:
```python
adf_cc = f"{ifgname}.adf.cc"
mask = get_water_mask(adf_cc, mwidth, lt, demw, demn, dempar)
if apply_water_mask:
    adf_cc, _ = water_mask_cc_mmli(adf_cc, mmli, int(mwidth), int(mlines), lt, int(demw), mask)

# produce the validity mask file f"{adf_cc_masked}_mask.bmp"
execute(f"rascc_mask {adf_cc} - {width} 1 1 0 1 1 0.1", uselogging=True)
```
(moving the `get_water_mask` call out of `get_masked_files`, which I've renamed to `water_mask_cc_mmli`)